### ISSUE TITLE
Should the training scripts include the noise shift?

### ISSUE THREAD

**Issue by xilanhua12138** [kohya-ss/musubi-tuner#54]
I noticed that in hunyuanvideo, the noise shift is relatively high: 7.0
should this shift be added into training process？

**Comment by Sarania** [comment]
I believe it is with the current recommended default settings:
```
--timestep_sampling shift --discrete_flow_shift 7.0 
```
It was previously using sigmoid/1.0, but kohya changed the sample command to that a bit over a week ago. He mentioned:

"Update: Changed the sample training settings to a learning rate of 2e-4, --timestep_sampling to shift, and --discrete_flow_shift to 7.0. Faster training is expected. If the details of the image are not learned well, try lowering the discete flow shift to around 3.0."

**Comment by Enyakk** [comment]
I am running comparison training tests now.

- My impression is that shift at flow shift 7.0 may be good for learning concepts. However it underperforms when learning specific photorealistic characters as features are not entirely picked up after a lengthy amount of training steps.
- Shift with flow shift 3.0 is not yet sufficiently tested
- Sigmoid at flow shift 1.0 excels at learning characters
- logit_normal may be another option for learning characters (not sufficiently tested)

**Comment by Sarania** [comment]
> * My impression is that `--timestep_sampling shift --discrete_flow_shift 7.0`  may be good for learning concepts. However it underperforms when learning specific photorealistic characters as features are not entirely picked up after a lengthy amount of training steps.

I concur with this. I have a photoreal character I'm trying to learn and with shift/7.0 it mostly learns her, but not the two dark freckles on the left side of her face that kind of identify her. Furthermore I have another video LORA that's focused on
<details>
  <summary>a slightly NSFW subject</summary>
 tongue kissing and it learns the overall theme and motion very well, but the actual detail fine of tongues interacting it does not, even when including higher resolution training buckets.
</details>

so I wonder if even for video it could still be beneficial to tweak the shift further. I will try sigmoid/1.0 for my character and see if it learns her freckles better!

**Comment by FurkanGozukara** [comment]
> * `--timestep_sampling sigmoid --discrete_flow_shift 1.0`

thank you for feedback. does it impact how you use after training @Enyakk

**Comment by FurkanGozukara** [comment]
@Sarania ty looking forward to it

**Comment by Sarania** [comment]
> I've currently got a run of my video dataset going at shift/3.0 (loss decreasing quite nicely) and I will update this post later when it finishes to say whether it improved fine detail pickup or not!

So... yes! It definitely learned the fine detail interactions better!

**Comment by LeonNerd** [comment]
Hello,sorry for asking another question here. I reproduced the PuLID method in sd3. In the first stage, I can output the image normally, but different id weights have different effects. I currently use 0.5. A larger weight will cause the image to be distorted. In the second stage, I added hypersd on the original basis. Here, the image is generated in 4 steps starting from pure noise. During the image generation process, the IDformer and cross attention modules have gradients, and the vae decoding tensor will also retain the gradients. The subsequent arcface feature extraction will also have gradients. IDformer and cross attention modules are embedded in hypersd, so my id loss will be updated with gradients in the hypersd branch, but I am curious why my training does not converge. I am not sure how much data or how many steps are needed to converge.
